QUARKUS-2489.md

```diff
@@ -0,0 +1,53 @@
+# QUARKUS-2489 OIDC Back channel logout
+
+Jira: https://issues.redhat.com/browse/QUARKUS-2489
+
+This feature is talking about logout flow on an OIDC provider as Keycloak
+
+## Scope of the testing
+
+Backchannel logout applies to `quarkus-oidc` and basically is a standard way to do logout in to a single sign-on service.
+We will use Keycloak as an OIDC provider and the flavor of the extensions (classic or reactive) is not relevant for this coverage. 
+
+Absolute Back-Channel Logout URL is calculated by adding `quarkus.oidc.back-channel-logout.path` to the current endpoint URL and then based on your configuration you could do a local and / or a global logout.
+
+We will create a valid token by an authorization flow and then after validating the token against some authenticated API we will proceed with the logout flow and verify the following constraints:
+- The logout response SHOULD include the Cache-Control HTTP response header field with a no-store value
+- Refresh tokens issued without the offline_access property to a session being logged out SHOULD be revoked. Refresh tokens issued with the offline_access property normally SHOULD NOT be revoked.
+- Logout token should be encrypted, decrypt it using the keys and algorithms that the Client specified during Registration. If logout Token is not encrypted, then reject it
+- Validate the Logout Token signature in the same way that an ID Token signature is validated
+- Validate the algorithm Header Parameter in the same way it is validated for ID Tokens
+- Validate the iss, aud, and iat Claims in the same way they are validated in ID Tokens
+- Verify that the Logout Token contains a sub Claim, a sid Claim, or both
+- Verify that the Logout Token contains an events Claim whose value is JSON object containing the member name http://schemas.openid.net/event/backchannel-logout
+- Verify that the Logout Token does not contain a nonce Claim.
+  
+If any of the validate logout tokens fails you must return an HTTP 400. Otherwise, proceed to perform the logout actions.
```

Comment:
Logout is async, only time you get 400 is that your token claims are invalid (e.g. missing back-channel claim) or expiration is missing or token is expired. We could check that as it makes sense to require valid token, but IMHO not very important as it's something like "pre-validaton", actual validation is done by Keycloak and we won't know about that except that we were not logged out. You decide :-)